pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -87,26 +85,59 @@ type AzVolumeAttachmentList struct {
 }
 
 // +genclient
-// +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// AzDriverNode is a specification for a AzDriverNode resource
+// AzDriverNode is a representation of a node, where azure CSI driver node plug-in runs.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="NodeName",type=string,JSONPath=`.spec.nodeName`,description="Name of the Node which this AzDriverNode object represents."
+// +kubebuilder:printcolumn:name="ReadyForVolumeAllocation",type=boolean,JSONPath=`.status.readyForVolumeAllocation`,description="Indicates if the azure persistent volume driver is ready for new pods which use azure persistent volumes."
+// +kubebuilder:printcolumn:name="LastHeatbeatTime",type=integer,JSONPath=`.status.lastHeatbeatTime`,description="Represents the time stamp at which azure persistent volume driver sent a heatbeat."
 type AzDriverNode struct {
 	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   AzDriverNodeSpec   `json:"spec"`
-	Status AzDriverNodeStatus `json:"status"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// spec defines the desired state of a AzDriverNode.
+	// Required.
+	Spec   AzDriverNodeSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+	// status represents the current state of AzDriverNode.
+	// If this is nil or empty, clients should prefer other nodes
+	// for persistent volume allocations or pod places for pods which use azure persistent volumes.
+	// +optional
+	Status *AzDriverNodeStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
 }
 
-// AzDriverNodeSpec is the spec for a AzDriverNode resource
+// AzDriverNodeSpec is the spec for a AzDriverNode resource.
 type AzDriverNodeSpec struct {
-	UnderlyingNode string `json:"underlyingNode"`
+	// Name of the node which this AzDriverNode represents.
+	// Required.
+	NodeName *string `json:"nodeName" protobuf:"bytes,1,opt,name=nodeName"`
 }
 
-// AzDriverNodeStatus is the status for a AzDriverNode resource
+// AzDriverNodeStatus is the status for a AzDriverNode resource.
 type AzDriverNodeStatus struct {
```

Comment:
Since the fields are all updated by the driver, why are they marked as optional?

Reply:
Please look at my comment above, and let me know what you think. I'll update the PR accordingly.